### DISCUSSION TITLE
Import/Index causes unspecified crash

### DISCUSSION THREAD

**Discussion by sarumont** [photoprism/photoprism#2832] [Q&A] [answered]
I have Photoprism running in a Kubernetes cluster (single node) via `k3s`. Importing images seems to be crashing the app. I have debug logging on, and there is nothing in the logs - the process just dies.

My system:
 - 4 cores, x86 (`i5-6500T`)
 - 16GB RAM

Photoprism (`photoprism/photoprism:220901-bullseye`):
 - allocated 6GB RAM via `k8s`
 - limited to a single worker (yes, I read the troubleshooting guide)

During the import, I see my machine hit a max of 3GB RAM utilized, about 8.5% utilized by Photoprism (which is 1.36GB), so it doesn't appear to be a resource issue, as I am well within the capacity of my hardware.

Any way to get additional debugging? There is nothing logged to the system journal nor to Photoprism's logs.

There is no consistency in how many images it will process before crashing (could be 10, could be 100). If I restart the import post-crash, it will move along just fine, so it's not choking on a particular image or type of image.

Edit: I should mention I was able to index about 3k files in my "Originals" before with no problem that I saw. This seems to be happening only on import, though I have not yet attempted a complete re-index.

**Comment by lastzero** [comment]
You could try our development preview. It includes a few fixes and improvements for indexing. However, I suspect this has something to do with Kubernetes and/or Linux kernel security. Hard to guess what specifically.

**Reply by sarumont** [reply]
I've ended up switching gears a bit wrt my homelab deployment, so I wouldn't be opposed to moving Photoprism into `docker`, as that seems to be the more paved path.

Just re-deployed the `preview` tag, though, and I'll test that now.

**Reply by sarumont** [reply]
Crashed again on `preview`. I'll try to reproduce in pure Docker to see if it's a k8s issue...

**Reply by lastzero** [reply]
Could be one of the RAW converters or FFmpeg unless you have JPEGs only. An extremely huge JPEG could also cause an out of memory error.

**Reply by lastzero** [reply]
There is a small chance that this is the problem that led to this behavior:
- https://github.com/golang/go/issues/56309

A new development preview based on Go 1.19.3 will be available for testing soon.

**Reply by sarumont** [reply]
> Could be one of the RAW converters or FFmpeg unless you have JPEGs only. An extremely huge JPEG could also cause an out of memory error.

It crashes on JPEGs and videos, and I have no RAW files. I'll look for the new preview with the updated Go version...

**Reply by lastzero** [reply]
It's available now. Happy testing! Thank you :)

**Reply by sarumont** [reply]
Seems to be crashing just the same, unfortunately. :( Going to try to get a docker container running on the same machine...

**Reply by lastzero** [reply]
You seem to be the only user affected, nobody else reporting anything similar.

**Reply by sarumont** [reply]
Yep...looking like it may be a quirk of running in Kubernetes. My import has been running solidly for the past 20 minutes in docker.

If you care about supporting k8s, I can try to dig in some more. I'm fine with it in docker myself, though.

**Reply by lastzero** [reply]
What means care about supporting? Many use it with Kubernetes, we even have contributed docs: https://docs.photoprism.app/getting-started/advanced/kubernetes/

**Reply by sarumont** [reply]
Meaning officially supporting it - which the documentation would seem to indicate you do. :)

I'll see if I can learn any more about what this crash is, then, as it may be useful to someone else.

**Reply by lastzero** [reply]
For sure. As I said, wasn't aware of any issues up until now. Many use it with Kubernetes.

**Reply by lastzero** [reply]
Could be the specific version or even a hardware instability. Let us know!

**Comment by lastzero** [comment]
@sarumont Did this happen when converting DNG/RAW files? If so, it could be the same problem as reported here:

- #2917

**It would be great everyone affected could figure out if there is an upstream issue with any of the RAW converters that needs to be fixed (and report it to the maintainers if necessary). Thank you very much!**

Also note:
- Darktable and Rawtherapee don't yet support ProRAW DNG files, so you might need to [blacklist the DNG extension](https://docs.photoprism.app/getting-started/config-options/#file-converters)  for both
- JPEGs can usually be extracted from DNG headers without processing, so a RAW converter is not strictly necessary
- if you are using Kubernetes with a custom config, you may be using different defaults than everyone else

**Reply by sarumont** [reply]
@lastzero Nope, I have no DNG or RAW files - just PNGs, JPEGs, and HEIC.

**Comment by sarumont** [comment, marked as answer]
I've revisited my k8s deployment and added MariaDB rather than SQLite. This configuration seems to be stable, so I'm going to attribute this crash to using SQLite over an NFS share (I've had other issues with SQLite DBs on network shares in the past).

**Reply by lastzero** [reply]
Lol, Sqlite over NFS definitely is a very bad idea.

**Reply by lastzero** [reply]
👉 https://docs.photoprism.app/getting-started/troubleshooting/sqlite/

**Reply by lastzero** [reply]
Updated our docs for you: https://github.com/photoprism/photoprism-docs/commit/1668a761b95f6468fe850337d1e4550007f043d8#diff-1c74b1a5c3f3cc9f1508d0035514abf1a338aa21082a657f645deaff28cb582a
